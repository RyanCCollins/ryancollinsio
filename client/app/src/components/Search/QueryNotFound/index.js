import React, { PropTypes } from 'react';
import Heading from 'grommet-udacity/components/Heading';
import cssModules from 'react-css-modules';
import styles from './index.module.scss';

const fontSizeMap = {
  small: 'h5',
  medium: 'h3',
  large: 'h1',
};

const QueryNotFound = ({ itemName, fontSize, onReset }) => (
  <div className={styles.noneFound}>
    <div className={styles.thinkingFace}>
      ðŸ¤”
    </div>
    <div className={styles.noneFoundText}>
      <Heading align="center" strong tag={fontSizeMap[`${fontSize}`]}>
        {`Sorry, but we couldn't find any ${itemName} that matched your query.`}
      </Heading>
      {onReset &&
        <Heading tag="h5" align="center">
          {'Suggestion: '}
          <a href="#" onClick={onReset}>reset filters</a>
          {' to see all projects.'}
        </Heading>
      }
    </div>
  </div>
);

QueryNotFound.propTypes = {
  itemName: PropTypes.string.isRequired,
  fontSize: PropTypes.oneOf(['small', 'medium', 'large']),
  onReset: PropTypes.func,
};

QueryNotFound.defaultProps = {
  fontSize: 'medium',
  hasResetLink: false,
  itemName: 'items',
};

export default cssModules(QueryNotFound, styles);
