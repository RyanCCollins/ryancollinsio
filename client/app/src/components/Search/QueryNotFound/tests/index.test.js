import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import QueryNotFound from '../index';

describe('<QueryNotFound />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <QueryNotFound />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
