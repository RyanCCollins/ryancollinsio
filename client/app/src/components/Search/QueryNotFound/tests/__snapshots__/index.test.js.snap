exports[`<QueryNotFound /> should render with default props 1`] = `
<div>
  <div>
    ðŸ¤”
  </div>
  <div>
    <Heading
      align="center"
      strong={true}
      tag="h3">
      Sorry, but we couldn\'t find any items that matched your query.
    </Heading>
  </div>
</div>
`;
